index.js

```diff
@@ -107,8 +107,7 @@ class ServerlessNestedPlugin {
 
                         this.serverless.service.provider.compiledCloudFormationTemplate = parentStack;
                         fs.writeFile(this.packagePath + '/compiled-cloudformation-template.json',
-                            JSON.stringify(parentStack, null, ' '));
-                        resolve();
+                            JSON.stringify(parentStack, null, ' '), () => { resolve(); });
```

Comment:
Do we actually need to move `resolve()` into the callback?

Reply:
JSON.stringify(parentStack, null, ' '), resolve);
This can probably work. However, I am no longer using this plugin already. I have revamped the lambda functions and shrink the no of objects.